### ISSUE TITLE
Make the Geocoris more easily available.

### ISSUE THREAD

**Issue by quyykk** [endless-sky/endless-sky#6338]
### Problem Description

Currently the only shipyard that sells the Geocoris (the best Hai Heavy Freighter) is on Hai Home. That planet is restricted and the only way to land on it is either through some mission or increasing your reputation with the Hai. The former isn't really practical and the latter requires you to either repair Hai ships or destroy Unfettered ships.

It seems weird that a Heavy Freighter is locked behind combat related things.

### Related Issue Links

Didn't find any.

### Desired Solution

Add a "The Hai trust you now" mission just like for the Coalition that gives you the required reputation to land on Hai Home if you complete enough jobs for the Hai.

### Alternative Approaches

Sell the Geocoris on other shipyards.

**Comment by Zitchas** [comment]
I'd personally like to see generic missions raising the reputation of the player with those factions that appreciate general trade. This was something that EV did, and I thought it was wonderful. I might be a rebel that the Confederation Navy will kill on sight, but the local civilian government just considers me a friendly merchant that has delivered a million credits worth of cargo and passenegers over the past year.

I seem to recall that letting jobs affect reputation was considered and ultimately rejected by MZ because it was "too farmable" and people used this to enable themselves to capture things like Carriers without making the Navy permanently hostile.

We still *can* do it, in fact. The code is there, it's just a simple addition to each job. 

The alternative solution, though is changing how reputation works. Instead of having it just go up and down by the value of the crew killed, modified by government variables, what if killing a ship of that government automatically dropped the rep to 0 and then started counting down from there? Or at least, have that be an option for some societies, anyway. Case in point: Reality. Doesn't matter how beloved a delivery driver you are, the day you brutally slaughter the security company employees and steal the bank truck full of cash in full view of witnesses, cameras, and the general public; That is the day you suddenly go from "Respected and well-liked member of the community" to "creepy central figure in a real-life horror story" and everyone is terrified of you.

**tldr**: The reputation system is broken; things that should affect your rep (like being an upstanding merchant) don't; things that *should* kill your rep have very weak impacts. The fact that there are a whole bunch of resets for various factions erasing your past at various points in the story makes it even more meaningless.

**Comment by Hurleveur** [comment]
I think that the way we can fix the reputation system is by making it closer to real life : if you trade it will probably make people like you a bit (which it should) but it will fade out over time. Boarding and killing people on a carrier and potentially shooting down the whole fleet nearby now that is a war crime and should be considered as such : you are now an enemy. So what I propose is :
-We keep the permanent relation with each faction, but add a temporary one.
-Doing small missions gives temporary relations and so does selling much needed cargo, that is highly expensive (like medicals in a war zone, metals to make much needed ships or luxury goods where they cost a lot, making people able to afford it). For the exact numbers and the depreciation of relations I don't know yet, I'm welcome to suggestions.
-As for the negative temporary, having illegal goods on you should apply a temporary negative until they finally shoot you on sight if you are known to do that for a living. (but they will eventually forget)
-If all you do is shooting them without disabling anyone and then jump out, it could be a mistake, or whatever, so you are forgiven (as is currently the case).
-However, should you disable a ship, from that point all temporary boosts are removed. Should you board it even the permanent ones will be removed, taking your relation down to 0 and then applying the normal reduction for killing people disabling ships etc on top of it. (because even if you saved Earth people will see you have changed now, and will no longer know what to think)
-You can still pay them to be able to land on any planet. But if you have negative relations any small missions you may do will only mean the navy temporarily forgive you (if you go back above 0) and you need to keep working on your relations to be able to not be shot on sight for the atrocities you committed. Or you would need to do say 10 times as much as you did to them to their enemies, or do the story missions after said atrocities (what is most recent is what is remembered).
-Any good relations will enable you to buy ships, outfits etc with all alien governments, that do not easily trust outsiders (the Hai may think you are a pirate, the remnants trust no one...), with a given amount being enough to buy the outfits, then the small ships, and then the bigger ones (with the Geocoris being sold on other shipyards obviously). Missions will still allow you to buy them on a permanent basis, but doing a lot of small things will make them trust you too. (tough never enough to buy ships that require a license)

They could even hire bounty hunters against you if the rep is low enough (but that would be for another PR probably ?)

PS : we could also make it so that all relations are temporary but I like the idea of story missions giving you permanent relation with factions because of how important they are, except if you declare war on them that relation will stay.

**Comment by Amazinite** [comment]
As of 5224680 you can now earn reputation with the Hai for running jobs.
